app/controllers/registration_controller.rb

```diff
@@ -0,0 +1,53 @@
+class RegistrationController < ApplicationController
+  def new
+    form_options
+  end
+
+  def create
+    form_options
+    @endpoint = registration_endpoint
+    @headers = registration_headers
+    @registration_args = registration_args
+    render 'show'
+  end
+
+  private
+
+  def form_options
+    @organizations = Organization.authorized(:view_organizations).select(:id, :name)
+    @locations = Location.authorized(:view_locations).select(:id, :name)
+    @host_groups = Hostgroup.authorized(:view_hostgroups).select(:id, :name)
+    @operating_systems = Operatingsystem.authorized(:view_operatingsystems).select(:id, :title)
+    @smart_proxies = SmartProxy.authorized(:view_smart_proxies).select(:id, :name)
+    @plugin_params = plugin_params
+  end
+
+  def plugin_params
+    Foreman::Plugin.all.map(&:allowed_registration_vars).flatten.compact.uniq
+  end
+
+  def registration_headers
+    jwt = User.current.jwt_token!(expiration: params[:jwt_expiration].to_i.hours.to_i)
+    "-H 'Authorization: Bearer #{jwt}'"
+  end
+
+  def registration_endpoint
+    if params['smart_proxy'].empty?
+      global_registration_url
+    else
+      "#{SmartProxy.authorized(:view_smart_proxies).find(params[:smart_proxy]).url}/register"
+    end
+  end
+
+  def registration_args
+    from_plugins = Hash[plugin_params.map { |key| [key, params[key]] }]
+    url = params['smart_proxy'].empty? ? '' : SmartProxy.authorized(:view_smart_proxies).find(params[:smart_proxy]).url
```

Comment:
Why do you need to pass the URL as a registration arg? I thought the `/register` on the Smart Proxy would ensure the correct template is provided and there is no need to do this on the Foreman side.

Reply:
See: https://github.com/theforeman/foreman/pull/8054